TempMail/utils/Email.swift

```diff
@@ -55,8 +59,10 @@ class Email : ObservableObject{
                 guard let data = data else {return}
                 //updates instance variable
                 DispatchQueue.main.async {
-                    guard let inbox = self.parseInboxData(data: data) else {return}
-                    self.inbox = inbox
+                    if self.email_addr == email_addr {
+                        guard let inbox = self.parseInboxData(data: data) else {return}
```

Comment:
Do this part async and not on the main thread. Don't want to block the main thread for some simple decoding that could've been done in the background. Since the decoding is going to be fast and resource light, it wouldn't matter if it ends up not being used.